adapters/handlers/grpc/v1/prepare_reply.go

```diff
@@ -32,7 +32,25 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
-func searchResultsToProto(res []interface{}, start time.Time, searchParams dto.GetParams, scheme schema.Schema, usesPropertiesMessage bool) (*pb.SearchReply, error) {
+type mapper interface {
+	NewPrimitiveValue(v interface{}, dt schema.DataType) (*pb.Value, error)
+	NewNestedValue(v interface{}, dt schema.DataType, parent schema.PropertyInterface, prop search.SelectProperty) (*pb.Value, error)
+	NewNilValue() *pb.Value
+}
+
+type Replier struct {
+	mapper  mapper
```

Comment:
why not simply use mapper here?

Reply:
I don't see the point of adding extra code (and clicks when navigating in the IDE) just to make it more similar. We are not planning on adding another mapper and if we would do that in the future we could still add an interface then?